#1 [internal] load build definition from Dockerfile
#1 sha256:57eaf985d8813612f12a690083c7dfaf03ff3feeb928c01cda0ac056ad06c85b
#1 transferring dockerfile: 724B done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 sha256:acc05ad0a433e900d1e90cdd6197758f8dec921c7d139b8c163d784e0b70a1f6
#2 transferring context: 2B done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/library/ubuntu:20.04
#3 sha256:cd9ed232facd22264e9246054f9c3d7a9c42bdb414a225d8f8215114646e640c
#3 DONE 0.2s

#4 [ 1/18] FROM docker.io/library/ubuntu:20.04@sha256:ed4a42283d9943135ed87d4ee34e542f7f5ad9ecf2f244870e23122f703f91c2
#4 sha256:8f291decb28dc3b4198102409bcdae24147c130a31cf61bcaa5d4e4fd9a825c1
#4 DONE 0.0s

#16 [internal] load build context
#16 sha256:aa83fd3f6df68c30fc30dd08d51a1cde717739314f4980e1f6157a39085e7dfa
#16 transferring context: 1.74MB 0.5s done
#16 DONE 0.5s

#14 [11/18] RUN python -m pip install cryptography
#14 sha256:499bf538b6dd15cdaf9257d92c9c8678ab35cdce53f1d6327420f034461b563b
#14 CACHED

#6 [ 3/18] RUN apt-get upgrade --yes
#6 sha256:779c22f0b850b34ff5680c564ea788ae6c91389b447d4b5f1bb822a13242cf75
#6 CACHED

#9 [ 6/18] RUN python -m pip install pandas pymongo
#9 sha256:3ef7d5d41d632169da7adbfd9008846d00fb40c20e9b9a8e6ab294158ef4aff6
#9 CACHED

#7 [ 4/18] RUN apt-get install python3.8 python-is-python3 python3-pip --yes
#7 sha256:38b9e6c0906fbfea62cda624c2ae04b89e78e154eb0ee8960d192c8d1e0739bc
#7 CACHED

#5 [ 2/18] RUN apt-get update
#5 sha256:a7a70b9babceb8aabe1e5debe7f6add7f37aaa1213c5441c6c0085a626201fca
#5 CACHED

#10 [ 7/18] RUN apt-get install mysql-client --yes
#10 sha256:2745679caadab19ab15193f5f34860567a42ca824388289d165fb5fc9b2c9909
#10 CACHED

#17 [13/18] COPY ./requirements.txt /dbknitter/requirements.txt
#17 sha256:d87cf3b025714502cb8758b430b908c3d0f3f0bf4b20d0d10507285b3837ed4f
#17 CACHED

#12 [ 9/18] RUN python -m pip install redis
#12 sha256:bc710fd748e3bf8b268f59d31cf9823bbf3042b3e870a0962542d19f1b412602
#12 CACHED

#15 [12/18] RUN python -m pip install direct_redis
#15 sha256:4a9224ed27dc2ccdcd002d83f5e5acbbb7f3b38d0640a6b30edc09206f230deb
#15 CACHED

#11 [ 8/18] RUN python -m pip install pymysql
#11 sha256:24a4f922451f7d6c267941842b4e28eb6084198928f20bb397974c4b54d0eb29
#11 CACHED

#13 [10/18] RUN python -m pip install cdifflib
#13 sha256:75cb4463443188da464d967d6712f9f6c294706c91b4e8e8e4a6dac0476097a2
#13 CACHED

#8 [ 5/18] RUN python -m pip uninstall -y bson
#8 sha256:757b36d72104bf3c38f7156b7e3af083e23ceb9300ff494868a5dccb37c183ce
#8 CACHED

#18 [14/18] RUN pip install -r /dbknitter/requirements.txt
#18 sha256:a353b97380490e960dc9c004c4dad3d43d0b7a94724a849ec3d5e7ec45cfe366
#18 CACHED

#19 [15/18] COPY . /dbknitter
#19 sha256:a49f0f814225bd6fe3da717bf85e9b42736ea05bd32bbd1cb62cb32215a29aa9
#19 DONE 1.5s

#20 [16/18] WORKDIR /dbknitter
#20 sha256:6c4ffb8dd0b8bd7751eb17c78df489e251d29f26ad5d15f20acdd2990e982d1c
#20 DONE 0.0s

#21 [17/18] RUN pip install .
#21 sha256:b4afc07a356f6afa4f5824c1e71f2faf2eb919c98217e92ef1cc56aad28eb9f6
#21 0.596 Processing /dbknitter
#21 2.379 Requirement already satisfied: openai in /usr/local/lib/python3.8/dist-packages (from dbknitter==0.1.0) (1.3.5)
#21 2.384 Requirement already satisfied: requests in /usr/local/lib/python3.8/dist-packages (from dbknitter==0.1.0) (2.31.0)
#21 2.387 Requirement already satisfied: pydantic<3,>=1.9.0 in /usr/local/lib/python3.8/dist-packages (from openai->dbknitter==0.1.0) (2.5.2)
#21 2.390 Requirement already satisfied: typing-extensions<5,>=4.5 in /usr/local/lib/python3.8/dist-packages (from openai->dbknitter==0.1.0) (4.8.0)
#21 2.391 Requirement already satisfied: distro<2,>=1.7.0 in /usr/local/lib/python3.8/dist-packages (from openai->dbknitter==0.1.0) (1.8.0)
#21 2.391 Requirement already satisfied: anyio<4,>=3.5.0 in /usr/local/lib/python3.8/dist-packages (from openai->dbknitter==0.1.0) (3.7.1)
#21 2.405 Requirement already satisfied: tqdm>4 in /usr/local/lib/python3.8/dist-packages (from openai->dbknitter==0.1.0) (4.66.1)
#21 2.409 Requirement already satisfied: httpx<1,>=0.23.0 in /usr/local/lib/python3.8/dist-packages (from openai->dbknitter==0.1.0) (0.25.2)
#21 2.414 Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.8/dist-packages (from requests->dbknitter==0.1.0) (3.3.2)
#21 2.415 Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.8/dist-packages (from requests->dbknitter==0.1.0) (2023.11.17)
#21 2.416 Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.8/dist-packages (from requests->dbknitter==0.1.0) (3.6)
#21 2.417 Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.8/dist-packages (from requests->dbknitter==0.1.0) (2.1.0)
#21 2.419 Requirement already satisfied: annotated-types>=0.4.0 in /usr/local/lib/python3.8/dist-packages (from pydantic<3,>=1.9.0->openai->dbknitter==0.1.0) (0.6.0)
#21 2.421 Requirement already satisfied: pydantic-core==2.14.5 in /usr/local/lib/python3.8/dist-packages (from pydantic<3,>=1.9.0->openai->dbknitter==0.1.0) (2.14.5)
#21 2.422 Requirement already satisfied: sniffio>=1.1 in /usr/local/lib/python3.8/dist-packages (from anyio<4,>=3.5.0->openai->dbknitter==0.1.0) (1.3.0)
#21 2.423 Requirement already satisfied: exceptiongroup; python_version < "3.11" in /usr/local/lib/python3.8/dist-packages (from anyio<4,>=3.5.0->openai->dbknitter==0.1.0) (1.2.0)
#21 2.424 Requirement already satisfied: httpcore==1.* in /usr/local/lib/python3.8/dist-packages (from httpx<1,>=0.23.0->openai->dbknitter==0.1.0) (1.0.2)
#21 2.427 Requirement already satisfied: h11<0.15,>=0.13 in /usr/local/lib/python3.8/dist-packages (from httpcore==1.*->httpx<1,>=0.23.0->openai->dbknitter==0.1.0) (0.14.0)
#21 2.429 Building wheels for collected packages: dbknitter
#21 2.429   Building wheel for dbknitter (setup.py): started
#21 2.546   Building wheel for dbknitter (setup.py): finished with status 'done'
#21 2.547   Created wheel for dbknitter: filename=dbknitter-0.1.0-py3-none-any.whl size=36782 sha256=428771c684ab5637f93f3b6847f148270c8a3ae204d26c12f113796ec5bb5bf0
#21 2.547   Stored in directory: /tmp/pip-ephem-wheel-cache-6at5aqud/wheels/57/d4/17/cb4286ba457866ca85744b2bfc3c96e280880f7d32d199b163
#21 2.547 Successfully built dbknitter
#21 2.600 Installing collected packages: dbknitter
#21 2.616 Successfully installed dbknitter-0.1.0
#21 DONE 3.1s

#22 exporting to image
#22 sha256:e8c613e07b0b7ff33893b694f7759a10d42e180f2b4dc349fb57dc6b71dcab00
#22 exporting layers
#22 exporting layers 1.5s done
#22 writing image sha256:78ac21e6ef6de14f5e7537a26db63b3bc86a863af01a30ed5eef91cd24a70c0d done
#22 naming to docker.io/library/client-image done
#22 DONE 1.5s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
#1 [internal] load build definition from Dockerfile
#1 sha256:fdc63aed479cfac377f2529ee8a5e45c087785d0f4854ecb24a3a5c6f1f3866c
#1 transferring dockerfile: 541B done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 sha256:9abb562b877c2cc9a1a4b0dbb16c48217212e55338de7849f2010e5ca0661dde
#2 transferring context: 2B done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/library/mysql:latest
#3 sha256:581f9497edc116620469bb65354591ecdce3dfcf818d078c5f13986cd11b3f12
#3 DONE 0.3s

#4 [1/1] FROM docker.io/library/mysql:latest@sha256:1773f3c7aa9522f0014d0ad2bbdaf597ea3b1643c64c8ccc2123c64afd8b82b1
#4 sha256:c295488cd3d66d4c67c58595f61de0477f2d5456131b15e001ad4fa51c315311
#4 CACHED

#5 exporting to image
#5 sha256:e8c613e07b0b7ff33893b694f7759a10d42e180f2b4dc349fb57dc6b71dcab00
#5 exporting layers done
#5 writing image sha256:42cffdc4fb0382b53e3176e27efdec78cc28b8f172709bf41d208cf285395c72 done
#5 naming to docker.io/library/mysql-image done
#5 DONE 0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
#1 [internal] load build definition from Dockerfile
#1 sha256:606cb77f91f4faf4021fe305767e55f3dc810489273a3ef5e6b85826a3a79030
#1 transferring dockerfile: 711B done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 sha256:2e3bf054f04e3e3383de5c745dc212f4557f5d78e4648119a827abff96dd45d8
#2 transferring context: 2B done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/mongodb/mongodb-community-server:6.0-ubi8
#3 sha256:db9fa2d3c04f6f47b7028de5a03cfa56ceb5b573185a7994e61708c70c2a55c7
#3 DONE 0.2s

#4 [1/2] FROM docker.io/mongodb/mongodb-community-server:6.0-ubi8@sha256:6b2289a866ad467299de456f872b2e6a77948c835377d2a61dce962e511b64d2
#4 sha256:a3309e74794f9c6c11f4e4dab176c6f626a458dc608a986a5aa1e0ec78484595
#4 DONE 0.0s

#5 [2/2] RUN python3 -m pip install pandas pymongo
#5 sha256:4c262285873f341ae337b4f4c6f3e906d35d1e6f169aa5d3bc69ad5d6ede9c96
#5 CACHED

#6 exporting to image
#6 sha256:e8c613e07b0b7ff33893b694f7759a10d42e180f2b4dc349fb57dc6b71dcab00
#6 exporting layers done
#6 writing image sha256:255db1fd9a40407f0a61e43228e272d4bcc1211f590735f6ad0a3bc29b9f929b done
#6 naming to docker.io/library/mongodb-image done
#6 DONE 0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
#1 [internal] load build definition from Dockerfile
#1 sha256:9f28ccde3bfddec3e949712b50787516d9b883b5242fefe92eb3953a46aea59b
#1 transferring dockerfile: 339B done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 sha256:b934037f8683044ada85bc1cd4c8222fbd62261a56bb962540f614c12d9cb9b3
#2 transferring context: 2B done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/library/redis:7.2.2-bookworm
#3 sha256:76d21ddd67c3b5083cfa98c46d64e994eb755bbb2cabe0b45bf1ab438b3ea244
#3 DONE 0.2s

#4 [1/6] FROM docker.io/library/redis:7.2.2-bookworm@sha256:4b0069ac29dd0ee5d3aec7ee34fa1034017a4c3a2c094751a2412cc609e6dea3
#4 sha256:2b3d8b9e7dd3596835f96d0a826fa018ce98773f6a60c06055ec101a53cc944c
#4 DONE 0.0s

#7 [4/6] RUN apt install -y build-essential libssl-dev libffi-dev python3-dev
#7 sha256:b8f6b1866594989e265224e447ea352f251e6023b78aacec8cf8c9b5aaa0fde2
#7 CACHED

#8 [5/6] RUN apt install -y python3.11-venv
#8 sha256:248ec744bb080b2c8cbce469ab62c2fd3d0b545262a232f0fec9c66190b06a5f
#8 CACHED

#5 [2/6] RUN apt update && apt -y upgrade
#5 sha256:b42a3e0cb7cac3cde566cd0c1088380f6ea2a8a91bef0ab724b067a25a85cd3f
#5 CACHED

#6 [3/6] RUN apt install -y python3 python3-pip
#6 sha256:e143120aff70fc59ecc5517e880a35a29862aece39eeb2f2b23b5faee9b14656
#6 CACHED

#9 [6/6] RUN python3 -m venv /venv && . /venv/bin/activate && pip install pandas redis direct_redis
#9 sha256:efe492789e071b3d7bcfced230528ea90fe616a058a4e9fa0d03f9b875c89c62
#9 CACHED

#10 exporting to image
#10 sha256:e8c613e07b0b7ff33893b694f7759a10d42e180f2b4dc349fb57dc6b71dcab00
#10 exporting layers done
#10 writing image sha256:41aba202651a21713d847fcabf93c416f71dfe3f2837c2345db06e52d0e17a53 done
#10 naming to docker.io/library/redis-image done
#10 DONE 0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
Network cloudlab_default  Creating
Network cloudlab_default  Created
Container cloudlab-redis-1  Creating
Container cloudlab-client-1  Creating
Container cloudlab-mysql-1  Creating
Container cloudlab-mongodb-1  Creating
Container cloudlab-client-1  Created
Container cloudlab-redis-1  Created
Container cloudlab-mysql-1  Created
Container cloudlab-mongodb-1  Created
